From d29edbf29eaf769363b8c873badea9635dad8193 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 4 Jul 2018 18:24:03 +0300
Subject: [PATCH 0659/1051] fix: dts: a8k: Add CP eMMC regulator and update
 device parameters

Add GPIO regulator for controlling CP0 eMMC voltage (3.3V/1.8V)
Update CP0 SDHCI parameters in A7K/A8K DTSI files.

Additionally remove device enable statements from DTSI files - devices
should be enabled in board DTS file.

Change-Id: I50da80f3b69b370894bbf6b5beba19c032b31347
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../boot/dts/marvell/armada-7040-db.dtsi      | 19 ++++++++++++++++---
 .../arm64/boot/dts/marvell/armada-8040-db.dts |  2 --
 .../boot/dts/marvell/armada-8040-db.dtsi      | 16 ++++++++++++++++
 3 files changed, 32 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index c50df47198f0..f9a5728430a9 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -34,6 +34,19 @@
 		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
 	};
 
+	cp0_vccq_sd0_reg: cp0_vccq_sd0 {
+		compatible = "regulator-gpio";
+		regulator-name = "cp0-vccq-sd0";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		gpios = <&expander0 15 GPIO_ACTIVE_HIGH>;
+		gpios-states = <0>;
+		states = <1800000 0x1
+			  3300000 0x0>;
+		enable-active-high;
+	};
+
 	cp0_usb3_0_phy: cp0-usb3-0-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp0_reg_usb3_0_vbus>;
@@ -46,7 +59,6 @@
 };
 
 &uart0 {
-	status = "okay";
 	pinctrl-0 = <&uart0_pins>;
 	pinctrl-names = "default";
 };
@@ -115,9 +127,10 @@
 };
 
 &cp0_sdhci0 {
-	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdhci_pins>;
 	bus-width = <4>;
-	no-1-8-v;
+	vqmmc-supply = <&cp0_vccq_sd0_reg>;
 	non-removable;
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index ee89d0e883e9..fddafd679812 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -374,6 +374,4 @@
 
 &cp0_sdhci0 {
 	status = "okay";
-	bus-width = <8>;
-	non-removable;
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 7d5e82d30674..c80d4ca20bc8 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -53,6 +53,19 @@
 		gpio = <&expander1 0 GPIO_ACTIVE_HIGH>;
 	};
 
+	cp0_vccq_sd0_reg: cp0_vccq_sd0 {
+		compatible = "regulator-gpio";
+		regulator-name = "cp0-vccq-sd0";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		gpios = <&expander0 15 GPIO_ACTIVE_HIGH>;
+		gpios-states = <0>;
+		states = <1800000 0x1
+			  3300000 0x0>;
+		enable-active-high;
+	};
+
 	cp1_usb3_0_phy: cp1-usb3-0-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp1_reg_usb3_0_vbus>;
@@ -127,7 +140,10 @@
 };
 
 &cp0_sdhci0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdhci_pins>;
 	bus-width = <4>;
+	vqmmc-supply = <&cp0_vccq_sd0_reg>;
 	non-removable;
 };
 
-- 
2.17.1

